### DISCUSSION TITLE
TRPC query from Zod refine()

### DISCUSSION THREAD

**Discussion by BrendanC23** [trpc/trpc#5606] [Q&A]
Is there a way to use TRPC inside of a Zod `refine()` call? I can't do `trpc.getUser.useQuery();` because refine isn't within a hook. Normally, I would use `trpc.useUtils()`, but that also can't be used because it is a hook.

```ts
const schema = z.object({
    name: z.string(),
}).superRefine(async (data) => {
    // Make TRPC call using data.name
});
```

What is the best way to make a TRPC call in this case?

**Comment by ivan-kleshnin** [comment]
How would it work with the FE? 

`refine` changes a type so you can't have 1-1 correspondance without moving the whole thing to the FE. In my experience with Zodios and TS-REST (they share schemas BE<->FE, unlike TRPC), it all becomes highly problematic: bundling, security, etc, even without `refine`. And with `refine` it's getting impossible IMO – you can't magically transfer a function enclosed over BE scope to the FE.

Also, any special `refine` handling would make TRPC more Zod-specific and it seems the library aims to stay validation layer agnostic.

Such сustom validation logic belongs to either handlers or middlewares.
